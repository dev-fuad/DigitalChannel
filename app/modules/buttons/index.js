export {default as ButtonsScreen} from './ButtonsScreen';
