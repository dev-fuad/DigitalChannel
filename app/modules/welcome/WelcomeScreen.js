/**
 * WelcomeScreen.js
 * Author: Fuad Mohd. Firoz
 * Created: December 05, 2020
 *
 * @flow
 */

import React from 'react';
import {Text, TextInput, View} from 'react-native';
import {useDispatch, useSelector} from 'react-redux';
import {updateName} from '../../store/actions/UserActions';
import styles from './WelcomeScreenStyles';

import type {UserState} from '../../types/store';

const WelcomeScreen: () => React$Node = () => {
  const dispatch = useDispatch();
  const name = useSelector((state: UserState) => state.name);

  return (
    <View style={styles.container}>
      <View style={styles.content}>
        <View>
          <Text style={styles.greeting}>Welcome,</Text>
          <TextInput
            autoFocus
            style={styles.input}
            placeholder="Enter your name"
            value={name}
            onChangeText={(text) => dispatch(updateName(text.trimStart()))}
          />
        </View>
      </View>
    </View>
  );
};

export default WelcomeScreen;
