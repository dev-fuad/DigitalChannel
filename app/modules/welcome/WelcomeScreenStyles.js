import {StyleSheet, Dimensions} from 'react-native';

const {width} = Dimensions.get('screen');

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  content: {
    ...StyleSheet.absoluteFillObject,
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  greeting: {
    fontSize: 24,
  },
  input: {
    marginLeft: 8,
    width: width * 0.6,
    height: 40,
    fontSize: 28,
  },
});

export default styles;
