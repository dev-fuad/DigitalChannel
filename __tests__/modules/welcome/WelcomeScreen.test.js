import {cleanup, fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {Provider} from 'react-redux';
import {createStore} from 'redux';
import {WelcomeScreen} from '../../../app/modules/welcome';
import reducer from '../../../app/store/reducers/UserReducer';

const store = createStore(reducer);

describe('WelcomeScreen', () => {
  let screen;
  beforeEach(() => {
    screen = render(
      <Provider store={store}>
        <WelcomeScreen />
      </Provider>,
    );
  });

  afterEach(cleanup);

  it('should have an input', () => {
    const input = screen.getByPlaceholderText('Enter your name');
    expect(input).toBeDefined();
  });

  it('should update name on text change', () => {
    const input = screen.getByPlaceholderText('Enter your name');
    fireEvent.changeText(input, 'Chris');
    const name = store.getState().name;
    expect(name).toBe('Chris');
  });

  it('should match snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
