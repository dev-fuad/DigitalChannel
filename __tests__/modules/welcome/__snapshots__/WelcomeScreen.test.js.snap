// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 0,
        "flex": 1,
        "justifyContent": "center",
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontSize": 24,
          }
        }
      >
        Welcome,
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Enter your name"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 28,
            "height": 40,
            "marginLeft": 8,
            "width": 450,
          }
        }
        underlineColorAndroid="transparent"
        value="Chris"
      />
    </View>
  </View>
</View>
`;
